conda/version.py

```diff
@@ -302,6 +302,10 @@ def __new__(cls, spec):
         self.spec = spec
         if isinstance(spec, tuple):
             self.match = self.all_match_ if spec[0] == 'all' else self.any_match_
+        elif spec.startswith("^") and spec.endswith("$"):
```

Comment:
How about something like this, which isn't quite python code exactly, and really sloppy even if it were


    elif re.match(r'(\^\S+?\$)', spec) as m:
        first = m.group()
        operator = spec[m.end()] if len(spec) > m.end() else None
        if operator is None:
            self.spec = first
            self.regex = re.compile(spec)
            self.match = self.regex_match_
        else:
            return VersionSpec((operator, tuple(first, spec[m.end()+1:])))